### PR TITLE
Add span-lite/0.6.0

### PR DESCRIPTION
**Pull request by flexferrum**
Specify library name and version:  **span-lite/0.6.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/span-lite/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+from conans import ConanFile, tools
+
+
+class SpanLite(ConanFile):
+    name = "span-lite"
+    
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/martinmoene/span-lite"
+    description = "span lite - A C++20-like span for C++98, C++11 and later in a single-file header-only library"
+    author = "Martin Moene"
```
Comment (danimtb) [inline recipes/span-lite/all/conanfile.py]: Author attributes indicate the author of the recipe. As Conan Center Index is a collaborative project, recipes will be maintained and improved by the community, so we feel that the authorship will be tracked in the github history of the repo
```suggestion
```

File: recipes/span-lite/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    topics = ("conan", "cpp98", "cpp11", "cpp14", "cpp17", "span", "span-implementations")
+    license = "BSL-1.0"
+    no_copy_source = True
+
```
Comment (danimtb) [inline recipes/span-lite/all/conanfile.py]:
```suggestion
```

File: recipes/span-lite/all/test_package/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
```
Comment (danimtb) [inline recipes/span-lite/all/test_package/conanfile.py]:
```suggestion
```

**Comment by uilianries** [comment]
Hi @danimtb! I think this recipe is good now! Could you please review it again?

**Pull request merged**